[FIX] website: show all visibly dependent fields

PallaviSrivastavaa opened the pull request:
Description of the issue/feature this PR addresses: When more than one field was visibly dependent on multiple checkboxes, Only the first visibly dependent field was displayed on selecting more than one checkbox.

Current behavior before PR: Only the first visibly dependent field was displayed on selecting more than one checkbox.

Desired behavior after PR is merged: All visibly dependent fields are displayed when more than one checkboxes are selected



---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

R0-00 reviewed (changes requested):
I cannot see the contains option for multiple checkbox. Add it.

R0-00 commented on addons/website/static/src/snippets/s_website_form/000.js:
```diff
@@ -637,8 +637,16 @@ odoo.define('website.s_website_form', function (require) {
-                const currentValueOfDependency = formData.get(dependencyName);
-                return this._compareTo(comparator, currentValueOfDependency, visibilityCondition, between);
+            	var currentValueOfDependency;
+		if(["contains","!contains"].includes(comparator))
```
The code works fine, but you should follow the coding style. For example the code inside the if block is not properly aligned.

You can use const instead of var since it was defined as const previously.
You could use the ternary (conditional) operator to replace the 'if else' statement.

               
                const currentValueOfDependency = ["contains", "!contains"].includes(comparator)
                    ? formData.getAll(dependencyName).join()
                    : formData.get(dependencyName);

PallaviSrivastavaa replied:
I have made the changes

R0-00 reviewed (changes requested):
Regarding the commit message
* You can make changes to your commit message. You could give a small description of what was implemented. You have not made changes to the existing equal to option but added contains option. Just one sentence to explain the issue and a small description of what you did.
* Similarly you can also change the commit message heading

And that will do it.

R0-00 reviewed (changes requested):
Regarding the commit message

* You must give a small idea about the changes you made. Just mention that you have used contains and not contains. No need of extra newlines in the commit message.
*  Similarly you could also change the commit message heading if required.

And that will do it.

PallaviSrivastavaa committed:
[FIX] website: displays all fields with visible dependencies

When more than one field was visbily dependent on multiple checkboxes

Only the first visibly dependent field was displayed

small modifications were made to 000.js and s_website_form.xml file

It was checked whether each field meets it's visibility dependency condition

'contains' and 'doesn't contains' functionalities were added

while ensuring that other comparators' functionality was unaffected

PallaviSrivastavaa commented:
I have made the specified changes

R0-00 commented:
No need for further changes.
